build/cloudbuild.yaml

```diff
@@ -0,0 +1,17 @@
+# Build the module.
+steps:
+- name: gcr.io/cloud-builders/go:debian
+  entrypoint: bash
+  args:
+  - -c
+  - |
+    set -ex
+    IFS=- read str1 rev <<< "${_GIT_TAG}"
+    make build REV=${rev}
+    cp bin/csi-proxy.exe bin/csi-proxy${_GIT_TAG}.exe
+substitutions:
+  _GIT_TAG: '12345'
+artifacts:
+  objects:
+    location: 'gs://k8s-artifacts-csi'
```

Comment:
This is the "prod" bucket which I believe we should only push to by promoting from staging. Is there a staging bucket we can use instead?

Reply:
before I confirmed with bartsmykla@ that there is no such process yet.

Reply:
Correct, there's no GCS promotion process yet. I talk about it a little here: https://github.com/kubernetes/release/issues/964#issuecomment-564727533

I wrote [`gh2gcs`](https://github.com/kubernetes/release/tree/master/cmd/gh2gcs) very recently, which supports uploading GitHub releases to a GCS bucket.

See the usage here: https://github.com/kubernetes/release/pull/1311

So if you publish release tarballs to GitHub, you'll be able to do something like this to upload to GCS:

```shell
$ time gh2gcs --log-level debug --org kubernetes-csi --repo csi-proxy --bucket k8s-artifacts-csi --release-dir release --tags v0.0.0,v0.0.1,v0.0.2
```

If you run the command w/o the `--tags` flag, it'll iterate over all releases and upload them.
You could add a step in the cloudbuild.yaml to do this.

Eventually, I'll going to update the tool to accept a yaml config, so you can add a config file somewhere, that looks roughly like this:

```yaml
- org: foo
  repo: bar
  tags: []
  gcsBucket: bucket-name
  releaseDir: release-dir
```

Let me know what you think!

Reply:
@justaugustus thank you very much for providing this tool. For now, I will follow other csi repo build step first. Later we will try to see how to improve it with your tool.

Reply:
I couldn't quite follow the discussion, but is the suggestion to put ci images under a subfolder called "dev/ci", and then "promote" releases to "release/"? That way it's easy for someone to easily distinguish what is a production ready image vs a dev image.